sklearn/ensemble/_stacking.py

```diff
@@ -101,30 +101,44 @@ def _concatenate_predictions(self, X, predictions):
         This helper is in charge of ensuring the predictions are 2D arrays and
         it will drop one of the probability column when using probabilities
         in the binary case. Indeed, the p(y|c=0) = 1 - p(y|c=1)
+
+        When `y` type is `"multilabel-indicator"` or `"multiclass-multioutput"`
+        and the method used is `predict_proba`, `preds` can be either a ndarray
+        of shape (n_samples, n_class) or for some estimators a list of ndarray.
+        This function will drop one of the probability column in this situation as well.
         """
         X_meta = []
         for est_idx, preds in enumerate(predictions):
-            # case where the estimator returned a 1D array
-            if preds.ndim == 1:
+            if isinstance(preds, list):
+                # `preds` is here a list of n_targets 2D ndarrays of
+                # `n_classes` columns. The k-th column contains the
+                # probabilities of the samples belonging the k-th class.
+                #
+                # Since those probabilities must sum to one for each sample,
+                # we can work with probabilities of `n_classes - 1` classes.
+                # Hence we drop the first column.
+                for pred in preds:
+                    X_meta.append(pred[:, 1:])
+            elif preds.ndim == 1:
+                # Some estimator return a 1D array for predictions
+                # which must be 2-dimensional arrays.
                 X_meta.append(preds.reshape(-1, 1))
+            elif (
+                self.stack_method_[est_idx] == "predict_proba"
+                and len(self.classes_) == 2
+            ):
+                # Remove the first column when using probabilities in
+                # binary classification because both features are perfectly
+                # collinear.
+                X_meta.append(preds[:, 1:])
```

Comment:
At that point, it's hard to know if `preds` have two or more features.
The indexing with `1:` suggest that there might be more than two features, but `"both"` suggests that there are only two.

Also is `"features"` referring to `preds`' columns here?

Reply:
Yes, this condition can only be met when `preds.shape[1] == 2` since we are dealing with the other scenarios above.

`"features"` refers to `preds` indeed, does this new comment: "`... because both features preds are perfectly collinear.`" make it more explicit?

Reply:
I think columns aren't collinear but linked by affine relation.

If I were to rewrite it with those information, I would use something closer to the comment in the first block, with the indexing using a simple index rather than a range:
```suggestion
                # Since the two probabilities must sum to one for each sample, we
                # can work with one of them. For convention, we prefer working with
                # the probabilities for the positive class (i.e. the second column).
                # Hence we drop the other probabilities (i.e. the first column).
                X_meta.append(preds[:, 1])
```

What do you think of this suggestion?

Reply:
I think this is clearer and way more user-friendly!

Reply:
Well, to be exact, this is not collinear in vector algebraic sense.

(To me, this is another example showing that machine learning and proper terminology definitely are orthogonal. :upside_down_face: )

Reply:
But this is the definition in statistics: https://en.wikipedia.org/wiki/Multicollinearity

Reply:
Ah, my bad!